src/core/server/plugins/types.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Observable } from 'rxjs';
+import { Type } from '@kbn/config-schema';
+
+import { CoreSetup, CoreStart } from '../';
+import { ConfigPath, EnvironmentMode } from '../config';
+import { LoggerFactory } from '../logging';
+
+export type PluginConfigSchema = Type<unknown> | null;
+
+/**
+ * Dedicated type for plugin name/id that is supposed to make Map/Set/Arrays
+ * that use it as a key or value more obvious.
+ *
+ * @public
+ */
+export type PluginName = string;
+
+/** @public */
+export type PluginOpaqueId = symbol;
+
+/**
+ * Describes the set of required and optional properties plugin can define in its
+ * mandatory JSON manifest file.
+ * @internal
+ */
+export interface PluginManifest {
```

Comment:
what is the main purpose of this file? as I understand it's to share types between client and server. `PluginManifest` exists on the server side only, doesn't it?

Reply:
Right I made it for sharing types, but I figured I'd place all types for the plugins service in here for consistency.

Reply:
>  I figured I'd place all types for the plugins service in here for consistency.

ok. Although I'd rather keep them closer to the usage place. It helps us to track unused types and dependencies between different modules/files.